### PR TITLE
Removed GoQuorum permissioning interop

### PR DESCRIPTION
**Pull request by NickSneo**
## PR description
Removed GoQuorum permissioning from Besu

## Fixed Issue(s)
https://github.com/hyperledger/besu/issues/5459

### CODE REVIEW

**Comment by NickSneo** [comment]
@macfarla Can you plz review?

**Review by macfarla — COMMENTED** [review]
Thanks @NickSneo - I think we can remove the isQuorum flag processing too, I don't think it's used for anything else. 
Also this needs to go into the 23.7 release so just need to make sure we don't merge it in too early :)

File: config/src/main/java/org/hyperledger/besu/config/JsonGenesisConfigOptions.java [hunk 1]
```diff
@@ -477,7 +472,6 @@ public Map<String, Object> asMap() {
 
     if (isQuorum()) {
```
[the remark this answers is not in the document]
Comment (NickSneo) [reply config/src/main/java/org/hyperledger/besu/config/JsonGenesisConfigOptions.java]: Removed isQuorum() too

**Review by macfarla — COMMENTED** [review]
looking good. what is the result if you start besu with a genesis file that still has isQuorum=true in it? we would need besu to just ignore it

**Comment by NickSneo** [comment]
> looking good. what is the result if you start besu with a genesis file that still has isQuorum=true in it? we would need besu to just ignore it

Rebased PR to latest main
Blockchain works fine with isQuorum=true in genesis file and ignores it

genesis file used - 
```
{
  "config" : {
    "chainId" : 1337,
    "homesteadBlock" : 0,
    "eip150Block" : 0,
    "eip150Hash" : "0x0000000000000000000000000000000000000000000000000000000000000000",
    "eip155Block" : 0,
    "eip158Block" : 0,
    "byzantiumBlock" : 0,
    "constantinopleBlock" : 0,
    "petersburgBlock" : 0,
    "istanbulBlock" : 0,
    "muirGlacierBlock" : 0,
    "berlinBlock" : 0,
    "londonBlock" : 0,
    "zeroBaseFee" : true,
    "arrowGlacierBlock" : 0,
    "grayGlacierBlock" : 0,
    "isQuorum" : true,
    "qip714block": 0,
    "contractSizeLimit" : 2147483647,
    "qbft" : {
      "blockperiodseconds" : 1,
      "epochlength" : 30000,
      "requesttimeoutseconds" : 2
    }
  },
  "nonce" : "0x0",
  "timestamp" : "0x58ee40ba",
  "gasLimit" : "0x1fffffffffffff",
  "difficulty" : "0x1",
  "mixHash" : "0x63746963616c2062797a616e74696e65206661756c7420746f6c6572616e6365",
  "coinbase" : "0x0000000000000000000000000000000000000000",
  "alloc" : {
    "fe3b557e8fb62b89f4916b721be55ceb828dbd73" : {
      "privateKey" : "8f2a55949038a9610f50fb23b5883af3b4ecb3c3bb792cbcefbd1542c692be63",
      "comment" : "private key and this comment are ignored.  In a real chain, the private key should NOT be stored",
      "balance" : "0xad78ebc5ac6200000"
    }
  },
  "extraData" : "0xf84fa00000000000000000000000000000000000000000000000000000000000000000ea94957e89815a200a3a62da447cb0f98898ecc18745947ee801898161f84fba714bdbb4a388ebd49273fdc080c0"
}
```

Result - 
```2023-07-13 03:28:48.467+05:30 | main | INFO  | Besu | Starting Besu
2023-07-13 03:28:49.390+05:30 | main | INFO  | Besu |
####################################################################################################
#                                                                                                  #
# Besu version 23.4.5-dev-23be7616                                                                 #
#                                                                                                  #
# Configuration:                                                                                   #
# Network: Custom genesis file specified                                                           #
# Network Id: 1337                                                                                 #
# Data storage: Forest                                                                             #
# Sync mode: Full                                                                                  #
# RPC HTTP APIs: ETH,NET,QBFT,WEB3,TRACE,DEBUG                                                     #
# RPC HTTP port: 8545                                                                              #
#                                                                                                  #
# Host:                                                                                            #
# Java: oracle_openjdk-java-19                                                                     #
# Maximum heap size: 6.00 GB                                                                       #
# OS: osx-aarch_64                                                                                 #
# Total memory: 24.00 GB                                                                           #
# CPU cores: 8                                                                                     #
#                                                                                                  #
####################################################################################################
2023-07-13 03:28:49.392+05:30 | main | INFO  | Besu | Connecting to 0 static nodes.
2023-07-13 03:28:49.392+05:30 | main | INFO  | Besu | Security Module: localfile
2023-07-13 03:28:50.128+05:30 | main | INFO  | Besu | Using the native implementation of alt bn128
2023-07-13 03:28:50.791+05:30 | main | INFO  | Besu | Using the native implementation of modexp
2023-07-13 03:28:50.792+05:30 | main | INFO  | Besu | Using the native implementation of the signature algorithm
2023-07-13 03:28:50.795+05:30 | main | INFO  | Besu | Using the Java implementation of the blake2bf algorithm
2023-07-13 03:28:50.808+05:30 | main | INFO  | RocksDBKeyValueStorageFactory | No existing database detected at /Users/nicks/Desktop/coding/besu/quorom-removed network/node1/data. Using version 1
2023-07-13 03:28:51.713+05:30 | main | INFO  | KeyPairUtil | Loaded public key 0x751a37184c9cfa0f03af35edaadfc07a408a7e85fd5de4211c1012a02a8b40561ef297a602a51215efa684b002c6d17c9fb17908adb5be85427d1f9e6e5a6993 from /Users/nicks/Desktop/coding/besu/quorom-removed network/node1/data/key
2023-07-13 03:28:51.778+05:30 | main | INFO  | ProtocolScheduleBuilder | Protocol schedule created with milestones: [GrayGlacier: 0]
2023-07-13 03:28:51.820+05:30 | main | INFO  | TransactionPoolFactory | Enabling transaction pool
2023-07-13 03:28:51.825+05:30 | main | INFO  | BesuControllerBuilder | TTD difficulty is not present, creating initial sync phase for PoW
2023-07-13 03:28:51.843+05:30 | main | INFO  | RunnerBuilder | Detecting NAT service.
2023-07-13 03:28:51.892+05:30 | main | INFO  | Runner | Starting external services ...
2023-07-13 03:28:51.892+05:30 | main | INFO  | JsonRpcHttpService | Starting JSON-RPC service on 127.0.0.1:8545
2023-07-13 03:28:51.935+05:30 | vert.x-eventloop-thread-1 | INFO  | JsonRpcHttpService | JSON-RPC service started and listening on 127.0.0.1:8545
2023-07-13 03:28:51.937+05:30 | main | INFO  | AutoTransactionLogBloomCachingService | Starting auto transaction log bloom caching service.
2023-07-13 03:28:51.937+05:30 | main | INFO  | LogBloomCacheMetadata | Lookup cache metadata file in data directory: /Users/nicks/Desktop/coding/besu/quorom-removed network/node1/data/caches
2023-07-13 03:28:51.941+05:30 | main | INFO  | Runner | Starting Ethereum main loop ...
2023-07-13 03:28:51.941+05:30 | main | INFO  | NatService | No NAT environment detected so no service could be started
2023-07-13 03:28:51.942+05:30 | main | INFO  | NetworkRunner | Starting Network.
2023-07-13 03:28:51.946+05:30 | nioEventLoopGroup-2-1 | INFO  | RlpxAgent | P2P RLPx agent started and listening on /[0:0:0:0:0:0:0:0]:30303.
2023-07-13 03:28:51.946+05:30 | main | INFO  | PeerDiscoveryAgent | Starting peer discovery agent on host=0.0.0.0, port=30303
2023-07-13 03:28:51.953+05:30 | vert.x-eventloop-thread-1 | INFO  | VertxPeerDiscoveryAgent | Started peer discovery agent successfully, on effective host=0:0:0:0:0:0:0:0%0 and port=30303
2023-07-13 03:28:51.954+05:30 | vert.x-eventloop-thread-1 | INFO  | PeerDiscoveryAgent | P2P peer discovery agent started and listening on /[0:0:0:0:0:0:0:0%0]:30303
2023-07-13 03:28:51.964+05:30 | vert.x-eventloop-thread-1 | INFO  | PeerDiscoveryAgent | Writing node record to disk. NodeRecord{seq=1, publicKey=0x03751a37184c9cfa0f03af35edaadfc07a408a7e85fd5de4211c1012a02a8b4056, udpAddress=Optional[/127.0.0.1:30303], tcpAddress=Optional[/127.0.0.1:30303], asBase64=-Je4QAL_muT70LVigZ73sEyeJQ81lAUDjSCMZgy9TGSP6ZGHMta7TUZQj5ErBYgOC-6fUgncJNWTq44cPHmNnHh4BtsBg2V0aMfGhCm7o52AgmlkgnY0gmlwhH8AAAGJc2VjcDI1NmsxoQN1GjcYTJz6DwOvNe2q38B6QIp-hf1d5CEcEBKgKotAVoN0Y3CCdl-DdWRwgnZf, nodeId=0xfbe8657bd3ad0b32a25d6cf37ee801898161f84fba714bdbb4a388ebd49273fd, customFields={tcp=30303, udp=30303, ip=0x7f000001, eth=[[0x29bba39d, 0x]], id=V4, secp256k1=0x03751a37184c9cfa0f03af35edaadfc07a408a7e85fd5de4211c1012a02a8b4056}}
2023-07-13 03:28:51.973+05:30 | main | INFO  | DefaultP2PNetwork | Enode URL enode://751a37184c9cfa0f03af35edaadfc07a408a7e85fd5de4211c1012a02a8b40561ef297a602a51215efa684b002c6d17c9fb17908adb5be85427d1f9e6e5a6993@127.0.0.1:30303
2023-07-13 03:28:51.973+05:30 | main | INFO  | DefaultP2PNetwork | Node address 0x7ee801898161f84fba714bdbb4a388ebd49273fd
2023-07-13 03:28:51.977+05:30 | main | INFO  | NetworkRunner | Supported capabilities: [eth/62, eth/63, eth/64, eth/65, eth/66, eth/67, eth/68], [istanbul/100], [snap/1]
2023-07-13 03:28:51.977+05:30 | main | INFO  | DefaultSynchronizer | Starting synchronizer.
2023-07-13 03:28:51.978+05:30 | main | INFO  | TransactionPoolFactory | Enabling transaction handling following initial sync
2023-07-13 03:28:51.978+05:30 | main | INFO  | FullSyncDownloader | Starting full sync.
2023-07-13 03:28:51.978+05:30 | main | INFO  | FullSyncTargetManager | Unable to find sync target. Currently checking 0 peers for usefulness
2023-07-13 03:28:51.985+05:30 | main | INFO  | Runner | Ethereum main loop is up.
2023-07-13 03:28:56.992+05:30 | EthScheduler-Timer-0 | INFO  | FullSyncTargetManager | Unable to find sync target. Currently checking 0 peers for usefulness
2023-07-13 03:29:01.998+05:30 | EthScheduler-Timer-0 | INFO  | FullSyncTargetManager | Unable to find sync target. Currently checking 0 peers for usefulness
2023-07-13 03:29:07.005+05:30 | EthScheduler-Timer-0 | INFO  | FullSyncTargetManager | Unable to find sync target. Currently checking 0 peers for usefulness
2023-07-13 03:29:12.007+05:30 | EthScheduler-Timer-0 | INFO  | FullSyncTargetManager | Unable to find sync target. Currently checking 0 peers for usefulness
2023-07-13 03:29:16.020+05:30 | nioEventLoopGroup-3-2 | INFO  | FullSyncTargetManager | Unable to find sync target. Currently checking 1 peers for usefulness
2023-07-13 03:29:21.022+05:30 | EthScheduler-Timer-0 | INFO  | FullSyncTargetManager | Unable to find sync target. Currently checking 1 peers for usefulness
2023-07-13 03:29:26.029+05:30 | EthScheduler-Timer-0 | INFO  | FullSyncTargetManager | Unable to find sync target. Currently checking 1 peers for usefulness
2023-07-13 03:29:31.036+05:30 | EthScheduler-Timer-0 | INFO  | FullSyncTargetManager | Unable to find sync target. Currently checking 1 peers for usefulness
2023-07-13 03:29:36.043+05:30 | EthScheduler-Timer-0 | INFO  | FullSyncTargetManager | Unable to find sync target. Currently checking 1 peers for usefulness
2023-07-13 03:29:41.046+05:30 | EthScheduler-Timer-0 | INFO  | FullSyncTargetManager | Unable to find sync target. Currently checking 1 peers for usefulness
2023-07-13 03:29:45.474+05:30 | BftProcessorExecutor-QBFT-0 | INFO  | QbftRound | Importing proposed block to chain. round=ConsensusRoundIdentifier{Sequence=1, Round=4}, hash=0xc6eef5b1c7d7f9a564738f95bcc3269a6309ac029d1c786cf407c3e4e94508d2
2023-07-13 03:29:45.481+05:30 | BftProcessorExecutor-QBFT-0 | INFO  | QbftBesuControllerBuilder | Produced #1 / 0 tx / 0 pending / 0 (0.0%) gas / (0xc6eef5b1c7d7f9a564738f95bcc3269a6309ac029d1c786cf407c3e4e94508d2)
2023-07-13 03:29:46.047+05:30 | BftProcessorExecutor-QBFT-0 | INFO  | QbftBesuControllerBuilder | Imported #2 / 0 tx / 0 pending / 0 (0.0%) gas / (0x059464938c00290e963bd4ffa8caf21ff7eb9774bcfa4edb1b383dafbe2711ba)
2023-07-13 03:29:47.033+05:30 | BftProcessorExecutor-QBFT-0 | INFO  | QbftBesuControllerBuilder | Produced #3 / 0 tx / 0 pending / 0 (0.0%) gas / (0x02e613f2ff480d89f8b67cd9ba5179d5b63694e3cf4eba2549816b3d19d80fae)```

**Review by macfarla — COMMENTED** [review]
overall looks good. prob needs a changelog entry since we announced deprecating it. and just need to make sure it goes in the right release ie 23.7 cc @non-fungible-nelson

**Comment by NickSneo** [comment]
> overall looks good. prob needs a changelog entry since we announced deprecating it. and just need to make sure it goes in the right release ie 23.7 cc @non-fungible-nelson

I have added the change log entry to 23.7.0 section

**Review by macfarla — APPROVED** [review]

**Pull request merged**